streams/order-stream.go

```diff
@@ -0,0 +1,51 @@
+package streams
+
+import (
+	"encoding/json"
+	"fmt"
+
+	"gopkg.in/jcelliott/turnpike.v2"
+)
+
+// TickerStream holds state for a Poloniex 'ticker' firehose
+type OrderStream struct {
+	broadcaster TickerBroadcaster
+	RecieveDone chan bool
+}
+
+// NewTickerStream connects to the Poloniex 'ticker' firehose.
+func NewOrderStream(market string) (err error) {
+	c, err := turnpike.NewWebsocketClient(turnpike.JSON, poloniexWebsocketAddress, nil)
+	if err != nil {
+		return fmt.Errorf("Unable to open websocket: %v", err)
+	}
+
+	_, err = c.JoinRealm(poloniexWebsocketRealm, nil)
+	if err != nil {
+		return fmt.Errorf("Unable to join realm: %v", err)
+	}
+
+	c.ReceiveDone = make(chan bool)
+
+	// TODO Unmarshal the proper ticker event
+	err = c.Subscribe(market, handleOrder())
+	if err != nil {
+		return fmt.Errorf("Subscription to %v failed: %v", poloniexWebsocketTopic, err)
+	}
+	return nil
+}
+
+func handleOrder() turnpike.EventHandler {
```

Comment:
Omg. Why do they use async patterns in Go. What is your take on this @frodebjerke? It's safe to say that I don't like it